### ISSUE TITLE
get arguments used to request certificate

### ISSUE THREAD

**Issue by brianjmurrell** [certbot/certbot#7428]
## My operating system is (include version):
CentOS 7.7

## I installed Certbot with (certbot-auto, OS package manager, pip, etc):
yum package manager

It would be useful to be able to ask `certbot` for the commandline that produced an existing certificate.

This would be useful if one wants to re-request the certificate with different options/parameters, like perhaps one wants to add more domain names to an existing certificate.

**Comment by adferrand** [comment]
Hello !

All arguments passed to Certbot to issue a certificate are saved in the renewal configuration located at : `/etc/letsencrypt/renewal/[YOURDOMAIN].conf`. This is actually used during the renewal process to execute Certbot with the same set of parameters than during the first issuance.

Then, if you want to extend an already issued certificate, it is possible, even if not recommended, to modify directly these configuration files: https://certbot.eff.org/docs/using.html#modifying-the-renewal-configuration-file. Upon modification, a `certbot renew` call will update existing certificates accordingly to these new parameters.

**Comment by brianjmurrell** [comment]
> All arguments passed to Certbot to issue a certificate are saved in the renewal configuration located at : /etc/letsencrypt/renewal/[YOURDOMAIN].conf

Indeed, I am aware of this.  The problem is that not everything in that file is an argument, AFAICT so one has to figure out which ones are and which arguments they are, etc.

I figured that since certbot knows the relationship between those values and it's options, it would be nice to have certbot be able to tell one what cmdline one used to generate the certificate for purposes of re-requesting.  I don't think operations like adding more domains to one's certificate can be achieved by modifying the renewal configuration files, can it?  If not, there's at least one use case where running a new `certbot` command is needed and thus knowing all of the arguments it used before would be useful.

**Comment by adferrand** [comment]
Yes I see. Indeed the SANs are extracted from the previous certificate by default. It is not possible to change this list from a `certbot renew` invokation + modifying the renewal configuration. The closest thing is to invoke `certbot certonly` to overwrite an existing certificate with a new configuration, but it will not take the previous one into account.

I see your request, but I doubt it is possible to implement something acceptable. You see, what is obtained in the renewal configuration is the result of a merge between initial command line parameters, environment variable, existing renewal configuration, and even command line parameters during the renew, because it is possible to do several things on the `certbot renew` command (like defining new hooks). Some options interact with each other, some are mutually exclusive.

Obtaining a bijective relation of the current renewal state to a set of command line parameter to reproduce it seems not possible for me at the moment.

**Comment by adferrand** [comment]
One slightest simple solution could be to set explicitly the list of domains to insert in a certificate in the renewal configuration. Then would open the possibility to edit the domains afterwards in a handy manner.

It still not a complete solution to retrieve a set of args to reproduce a given renewal configuration, but it would cover at least the major use case. 

Anyway, I put this issue in wishlist with proper flags if someone is willing to propose a PRs.

Thanks again for your report @brianjmurrell!
